lib/backpex/resource.ex

```diff
@@ -18,344 +19,72 @@ defmodule Backpex.Resource do
 
-    from(item in schema, as: ^schema_name, distinct: field(item, ^id_field))
-    |> item_query.()
-    |> maybe_join(associations)
-    |> maybe_preload(associations, fields)
-    |> maybe_merge_dynamic_fields(fields)
-    |> where_id(schema_name, id_field, id_type, id)
-  end
-
-  defp where_id(query, schema_name, id_field, :id, id) do
-    case Ecto.Type.cast(:id, id) do
-      {:ok, valid_id} -> where(query, [{^schema_name, schema_name}], field(schema_name, ^id_field) == ^valid_id)
-      :error -> raise Ecto.NoResultsError, queryable: query
-    end
-  end
-
-  defp where_id(query, schema_name, id_field, :binary_id, id) do
-    case Ecto.UUID.cast(id) do
-      {:ok, valid_id} -> where(query, [{^schema_name, schema_name}], field(schema_name, ^id_field) == ^valid_id)
-      :error -> raise Ecto.NoResultsError, queryable: query
-    end
-  end
-
-  defp where_id(query, schema_name, id_field, _id_type, id) do
-    where(query, [{^schema_name, schema_name}], field(schema_name, ^id_field) == ^id)
+    adapter.get!(primary_value, assigns, live_resource)
   end
 
   @doc """
-  Deletes the given record from the database.
-  Additionally broadcasts the corresponding event, when PubSub config is given.
+  Gets a database record with the given `fields` by the given  `primary_value`.
+
+  Returns `nil` if no result was found.
 
   ## Parameters
 
-  * `item` (struct): The item to be deleted.
-  * `repo` (module): The repository module.
-  * `pubsub` (map, default: `nil`): The PubSub config to use for broadcasting events.
+  * `primary_value`: The identifier for the specific item to be fetched.
+  * `assigns` (map): The current assigns of the socket.
+  * `live_resource` (module): The `Backpex.LiveResource` module.
   """
-  def delete(item, repo, pubsub \\ nil) do
-    item
-    |> repo.delete()
-    |> broadcast("deleted", pubsub)
+  def get(primary_value, assigns, live_resource) do
+    adapter = live_resource.config(:adapter)
+
+    adapter.get!(primary_value, assigns, live_resource)
   end
 
   @doc """
-  Deletes multiple items from a given repository and schema.
-  Additionally broadcasts the corresponding event, when PubSub config is given.
+  Deletes multiple items.
+  Additionally broadcasts the corresponding event for each deleted item.
 
   ## Parameters
 
   * `items` (list): A list of structs, each representing an entity to be deleted. The list must contain items that have an `id` field.
-  * `repo` (module): The repository module.
-  * `schema` (module): The Ecto schema module corresponding to the entities in `items`.
-  * `pubsub` (map, default: `nil`): The PubSub config to use for broadcasting events.
+  * `live_resource` (module): The `Backpex.LiveResource` module.
   """
-  def delete_all(items, repo, schema, pubsub \\ nil) do
-    id_field = EctoUtils.get_primary_key_field(schema)
+  def delete_all(items, live_resource) do
+    adapter = live_resource.config(:adapter)
+    pubsub = live_resource.config(:pubsub)
```

Comment:
This does not work because the `pubsub` config only returns the PubSub name / module, e.g., `Demo.PubSub`, but we also need the `topic` and `event_prefix`. Therefore, PubSub doesn't work at all at the moment.

Reply:
Fixed. I changed the PubSub configuration to a Keyword list, so all options (name, event_prefix, topic) are now available by calling `live_resource.config(:pubsub)`.